homeassistant/components/asuswrt/config_flow.py

```diff
@@ -76,25 +75,28 @@ def _show_setup_form(self, user_input=None, errors=None):
         if user_input is None:
             user_input = {}
 
+        schema = {
+            vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
+            vol.Required(CONF_USERNAME, default=user_input.get(CONF_USERNAME, "")): str,
+            vol.Optional(CONF_PASSWORD)
+            if self.show_advanced_options
```

Comment:
We have this check below. Please consolidate it below instead. We prefer not using ternary operator expression if if spans more than one line. Set password as required here, and below if advanced options set it as optional.

Reply:
The only difference is that this will move the field `Password` in the form below the field `Protocol`. If you think this is not a problem I can create a PR to implement the change.  Or it should be better repeat all fields definition for both condition to maintain the fields order?

Reply:
Why would it change the order? We'll set the item here and below, just with different attributes.

Reply:
I'm not sure that I exactly understand what you mean, anyway I created a follow up PR for this request.